tests/h/services/group_test.py

```diff
@@ -126,6 +124,57 @@ def test_create_open_group_description_defaults_to_None(self, service):
 
         assert group.description is None
 
+    def test_create_resrticted_group_returns_group(self, service, users):
+        creator = users['cazimir']
+
+        group = service.create_restricted_group(name='test_group',
+                                                userid=creator.username,
+                                                origins=['https://biopub.org'],
+                                                description='test_description')
+
+        assert group.name == 'test_group'
+        assert group.authority == 'example.com'
+        assert group.creator == creator
+        assert group.description == 'test_description'
+        assert group.joinable_by is None
+        assert group.readable_by == ReadableBy.world
+        assert group.writeable_by == WriteableBy.members
+        assert creator in group.members
+
+    def test_create_restricted_group_sets_scopes(self, service, matchers, users):
+        origins = ['https://biopub.org', 'http://example.com', 'https://wikipedia.com']
+
+        group = service.create_restricted_group(name='test_group',
+                                                userid=users['cazimir'].username,
+                                                origins=origins,
+                                                description='test_description')
+
+        assert group.scopes == matchers.unordered_list([
+            GroupScopeWithOrigin(h) for h in origins])
+
+    def test_create_restricted_group_always_creates_new_scopes(self, db_session, factories, service, users, matchers):
+        # It always creates a new scope, even if a scope with the given origin
+        # already exists (this is because a single scope can only belong to
+        # one group, so the existing scope can't be reused with the new group).
+        origins = ['https://biopub.org', 'http://example.com']
+        scopes = [factories.GroupScope(origin=h) for h in origins]
+
+        group = service.create_restricted_group(name='test_group',
+                                                userid=users['cazimir'].username,
+                                                origins=origins,
+                                                description='test_description')
+
+        for scope in scopes:
+            assert scope not in group.scopes
+
+    def test_create_restricted_group_description_defaults_to_None(self, service):
+        # Create a restricted group with no `description` argument.
+        group = service.create_restricted_group(name='test_group',
+                                                userid='cazimir',
+                                                origins=['https://biopub.org'])
+
+        assert group.description is None
```

Comment:
I think a couple more tests for `create_restricted_group()` are needed:

* Test that it sets the group's creator as a member (similar to the existing `test_create_private_group_adds_group_creator_to_members()`)
* Test that it publishes a "group-join" event (similar to `test_create_private_group_publishes_join_event()`)